vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -463,4 +469,16 @@ synchronized protected boolean hasNextContextHandler(RoutingContextImplBase cont
   synchronized protected boolean hasNextFailureHandler(RoutingContextImplBase context) {
     return context.currentRouteNextFailureHandlerIndex() < failureHandlers.size();
   }
+
+  private void handleDecodingError(RoutingContextImplBase context, Throwable e) {
+    if (router.getDecoderErrorHandler() == null)
+      context.response()
```

Comment:
Currently router has errorHandler that's pretty intuitive to handle any type of errors. What if instead setting up response right away here, instead context was set as failed, e.g.: just `content.fail(400)` -> so no need for decodeErrorHandler.
The status message is going to be resolved in the error handler as it's currently being done by ErrorHandlerImpl